internal/controller/rhdh/backstage.go

```diff
@@ -36,121 +34,166 @@ var ConfigMapNameAndConfigDataKey = map[string]string{
-	pluginsDetails orchestratorv1alpha1.RHDHPlugins,
-	clusterDomain string,
+	rhdhConfig orchestratorv1alpha1.RHDHConfig,
+	argoCDEnabled, tektonEnabled bool,
+	clusterDomain, wfNamespace string,
 	ctx context.Context, client client.Client) error {
 	bsLogger := log.FromContext(ctx)
 
 	bsLogger.Info("Handling Backstage resources")
 
-	bsConfigMapList := GetConfigmapList(ctx, client, clusterDomain, operator, pluginsDetails)
+	bsConfigMapList := GetConfigmapList(ctx, client, clusterDomain, wfNamespace, argoCDEnabled, tektonEnabled, rhdhConfig)
 
-	if err := client.Get(ctx, types.NamespacedName{
-		Namespace: operator.Subscription.TargetNamespace,
-		Name:      BackstageCRName,
-	}, &rhdh.Backstage{}); apierrors.IsNotFound(err) {
-		secret := rhdh.ObjectKeyRef{
-			Name: operator.SecretRef.Name,
-		}
-		backstageCR := &rhdh.Backstage{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: BackstageAPIVersion,
-				Kind:       BackstageKind,
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      BackstageCRName,
-				Namespace: operator.Subscription.TargetNamespace,
-				Labels:    operations.AddLabel(),
-			},
-			Spec: rhdh.BackstageSpec{
-				Application: &rhdh.Application{
-					AppConfig:                   &rhdh.AppConfig{ConfigMaps: bsConfigMapList},
-					DynamicPluginsConfigMapName: AppConfigRHDHDynamicPluginName,
-					ExtraEnvs: &rhdh.ExtraEnvs{
-						Secrets: []rhdh.ObjectKeyRef{secret},
+	rhdhNamespace := rhdhConfig.RHDHNamespace
+	rhdhName := rhdhConfig.RHDHName
+
+	if err := client.Get(ctx, types.NamespacedName{Namespace: rhdhNamespace, Name: rhdhName}, &rhdhv1alpha2.Backstage{}); err != nil {
+		if apierrors.IsNotFound(err) {
+			secret := rhdhv1alpha2.ObjectKeyRef{
+				Name: BackendAuthSecretName,
+			}
+			backstageCR := &rhdhv1alpha2.Backstage{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: BackstageAPIVersion,
+					Kind:       BackstageKind,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      BackstageCRName,
+					Namespace: rhdhConfig.RHDHNamespace,
+					Labels:    kubeoperations.AddLabel(),
+				},
+				Spec: rhdhv1alpha2.BackstageSpec{
+					Application: &rhdhv1alpha2.Application{
+						AppConfig:                   &rhdhv1alpha2.AppConfig{ConfigMaps: bsConfigMapList},
+						DynamicPluginsConfigMapName: AppConfigRHDHDynamicPluginName,
+						ExtraEnvs: &rhdhv1alpha2.ExtraEnvs{
+							Secrets: []rhdhv1alpha2.ObjectKeyRef{secret},
+						},
+						Replicas: util.MakePointer(BackstageReplica),
 					},
-					Replicas: util.MakePointer(BackstageReplica),
 				},
-			},
-		}
-		if err := client.Create(ctx, backstageCR); err != nil {
-			bsLogger.Error(err, "Error occurred when creating Backstage resource")
-			return err
+			}
+			if err := client.Create(ctx, backstageCR); err != nil {
+				bsLogger.Error(err, "Error occurred when creating Backstage resource", "CR-Name", rhdhName)
+				return err
+			}
+			bsLogger.Info("Successfully created Backstage resource", "CR-Name", rhdhName)
```

Comment:
instead of Backstage, we should use `RHDH` or `Red Hat Developer Hub` IMO, and this everywhere